tools/documentation_crawler/documentation_crawler/spiders/common/spiders.py

```diff
@@ -168,4 +169,10 @@ def error_callback(self, failure: Failure) -> Optional[Union[Failure, Iterator[R
                 return self.retry_request_with_get(response.request)
             self.logger.error("Please check link: %s", response.request.url)
 
+        if failure.check(TimeoutError):
```

Comment:
I think I'd prefer to just add chat.zulip.org to the set of sites that we exclude from testing (`EXCLUDED_URLS`).

Reply:
(If chat.zulip.org down and 500ing, we don't want CI to fail)

Reply:
The test-documentation script was exiting with TimoutError.  I am not sure whether this will come under HttpError. 

https://chat.zulip.org/#narrow/stream/43-automated-testing/topic/.2E.2Ftools.2Ftest-documentation.20failing

Or do you mean to just check whether the URL is present in EXCLUDED_URLS in case of TimeoutError as well.

Reply:
No, I mean we shouldn't try to check chat.zulip.org URLs at all, which is what adding it to EXCLUDED_URLS would achieve.

Reply:
(Or maybe that isn't how we skip things, hmm)

Reply:
Ahh, `_is_external_link` is what we'd change to consider `chat.zulip.org` external.

Reply:
Okay. That makes sense. So you don't want to make any requests to CZO at all in the first place.

So there are a lot of links in our docs that points to various streams
```
https://chat.zulip.org/#narrow/stream/7-test-here
https://chat.zulip.org/#narrow/stream/95-new-members
https://chat.zulip.org/#narrow/stream/1-announce
https://chat.zulip.org/#narrow/stream/137-feedback
https://chat.zulip.org/#narrow/stream/101-design
https://chat.zulip.org/#narrow/stream/138-user-community
https://chat.zulip.org/#narrow/stream/31-production-help
https://chat.zulip.org/#narrow/stream/95-new-members
https://chat.zulip.org/#narrow/stream/49-development-help
https://chat.zulip.org/#narrow/stream/44-git-help
https://chat.zulip.org/#streams/all
https://chat.zulip.org/#narrow/stream/65-checkins
https://chat.zulip.org/#narrow/stream/49-development-help
https://chat.zulip.org/#narrow/stream/91-code-review
https://chat.zulip.org/#narrow/stream/19-documentation
https://chat.zulip.org/#narrow/stream/58-translation
https://chat.zulip.org/#narrow/stream/92-learning
https://chat.zulip.org/#narrow/stream/48-mobile
https://chat.zulip.org/#narrow/stream/16-desktop
https://chat.zulip.org/#narrow/stream/127-integrations
https://chat.zulip.org/#narrow/stream/9-issues
https://chat.zulip.org/#narrow/stream/31-production-help
https://chat.zulip.org/#narrow/stream/31-production-help
https://chat.zulip.org/#narrow/stream/31-production-help
https://chat.zulip.org/
https://chat.zulip.org/help/
https://chat.zulip.org/#narrow/stream/49-development-help
https://chat.zulip.org/help/
https://chat.zulip.org/#narrow/stream/58-translation
https://chat.zulip.org/#narrow/stream/1-announce
https://chat.zulip.org/stats
https://chat.zulip.org
https://chat.zulip.org
https://chat.zulip.org/#narrow/stream/21-provision-help
```

We need to keep in mind to update these links if we rename a stream. Someone running `./tools/test-all` should also figure it out eventually even if we miss it anyway.

Reply:
Our spider already doesn’t validate fragments on external URLs (as determined by `_is_external_url`, not `_is_external_link`). And there’s no way it could tell whether those CZO fragments are valid even if it wanted to, without (a) being logged in to CZO, (b) simulating enough of a browser and running enough JavaScript to figure out what happens when you visit those fragments, and (c) having some unclear heuristic for what counts as a success.

Reply:
Yeah. Good point. Maybe we can just fetch the list of streams using the API and see if the fragments are valid?